Yoast/wordpress-seo
# Add upsell for Local SEO to search appearance

## Summary

This PR can be summarized in the following changelog entry:

* [not-for-changelog] Adds an upsell for Local SEO to search appearance page when selecting an organization in the knowledge graph settings.

## Relevant technical choices:

* Using PHP to get the background image URL because:
	- When using the import it goes through `react-svg-loader`, which turns it into a component.
    - Now it has the added advantage of not being bundled in the JS too. Thanks Alexander!
* Using just one image as background, not a fancier 9 sliced version, as this does not really change size too much. Thanks Sjardo!
* The thing I'm not 100% on right now is the empty parent div that will always be put on the page (id: `wpseo-local-seo-upsell`). Then the component is mounted on that depending on whether we want the upsell there or not. I left this because adding more logic in the PHP view is not really nice either. If you have a strong opinion about this, please share.

## Test instructions

This PR can be tested by following these steps:

_Viewing the upsell._
* Go to Search Appearance → General and select `Organization` as representation under the `Knowledge Graph` setting.
* See the screenshot as to how it should look. See the issue as to how it was designed.

_Install the helper plugin to manipulate the license_
* See the comment below for the instructions: https://github.com/Yoast/wordpress-seo/pull/12646#issuecomment-483181679
* The license information is cached, to refresh this you should visit the `SEO -> Premium` page (see Irene's screenshot). Which is also where you can see the current status.

_The upsell should only show when you do not have an active Local SEO license._
* ~~This is a bit tricky as it depends on which licenses are active on your site.~~
  - ~~You can create another vagrant box to test a site without a license.~~
  - ~~Or you can edit the PHP code. In `admin/class-config.php` on line 143 change the value of `showLocalSEOUpsell` into `true` or `false`.~~
* Set the `expiryDate` in the helper plugin to a date in the past (`2019-01-27T01:00:00.000Z`).
* Refresh the `SEO -> Premium` page to see if the status of Local SEO is indeed `NOT ACTIVATED`.
* Ensure the upsell is shown.
* Set the `expiryDate` in the helper plugin to a date in the future (`2019-12-27T01:00:00.000Z`).
* Refresh the `SEO -> Premium` page to see if the status of Local SEO is indeed `ACTIVATED`.
* Ensure the upsell is NOT shown.

_The upsell should only show when you are in Free, never in Premium._
* First check if you see the upsell in Free.
* Then [merge this branch into premium](https://github.com/Yoast/Wiki/wiki/Merging-Free-into-Premium) to check if it is indeed not there anymore.

_There is RTL support._
* Change to a right-to-left (RTL) language in WordPress.
* Ensure the upsell image is on the right and the background image is flipped horizontally.

_The link should contain UTM tags and other info_
* Ensure the link goes to `https://yoast.com/wordpress/plugins/local-seo/#utm_source=yoast-seo&utm_medium=software&utm_term=local-seo-kb-notification&utm_content=settings&utm_campaign=wordpress-ad`
* Ensure there is extra information passed like `php_version` and `platform_version`.

## Screenshot
<img width="668" alt="local_seo_upsell_screenshot" src="https://user-images.githubusercontent.com/35524806/55868565-fffb9800-5b84-11e9-97cf-b792361a9a04.png">

## UI changes

## Quality assurance


Fixes https://github.com/Yoast/wordpress-seo-premium/issues/2127
